integration_tests/union_02.py

```diff
@@ -0,0 +1,48 @@
+from ltypes import i32, f64, i64, dataclass, ccall, Union
+
+@dataclass
+class A:
+    ax: i32
+    ay: f64
+
+@dataclass
+class B:
+    bx: i64
+    by: f64
+
+@dataclass
+class C:
+    cx: i64
+    cy: f64
+    cz: f64
+
+@ccall
+class D(Union):
+    a: A
+    b: B
+    c: C
```

Comment:
Let's also test (probably in a new test file) a Union that is not c interoperable:
```python
class D(Union):
    dx: i64
    dy: f64
    dz: f64
```
Btw, I think this union feature can also be used for common blocks and equivalence in LFortran, see https://github.com/lfortran/lfortran/issues/718.